packages/extension/src/components/molecules/ChainIndicator/ChainIndicator.tsx

```diff
@@ -0,0 +1,84 @@
+import { FC, useState } from 'react';
+
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import Button from '@mui/material/Button';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+
+import { Chain } from '@gemwallet/constants';
+
+import { useNetwork } from '../../../contexts';
+import { Xahau, Xrp } from '../../atoms';
+import { LoadingOverlay } from '../../templates';
+
+export const ChainIndicator: FC = () => {
+  const { chainName, switchChain } = useNetwork();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const open = Boolean(anchorEl);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+    setIsLoading(false);
+  };
+
+  const handleChangeChain = (chain: Chain) => {
+    setIsLoading(true);
+    switchChain(chain).then(() => {
+      handleClose();
+    });
+  };
+
+  const renderChainIcon = (chain: Chain) => {
+    switch (chain) {
+      case Chain.XRPL:
+        return <Xrp style={{ width: '24px', height: '24px' }} />;
+      case Chain.XAHAU:
+        return <Xahau style={{ width: '24px', height: '24px' }} />;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div>
+      <Button
+        aria-controls="chain-menu"
+        aria-haspopup="true"
+        onClick={handleClick}
+        sx={{
```

Comment:
It would be nice to extract this style to a constant that you would define outside: `ChainIndicator`.
Eg. in a variable `const styles = {button: {yourStyle}}` doing so would reference your style variable and not create a new instance on each render.
See this as a micro-optimization :)